### ISSUE TITLE
Native OS X support

### ISSUE THREAD

**Issue by Frooxius** [Neos-Metaverse/NeosPublic#2764]
### Is your feature request related to a problem? Please describe.

Neos currently doesn't run on the OS X system natively. We're creating this issue to gauge interest in this being implemented at some point - please vote on this issue if you'd be interested in this!

### Describe the solution you'd like

Native OS X support. This will require compiling all of our native dependencies for OS X (there's some progress on those being moved to CI, which makes it a bit easier to setup) and solving any platform-specific issues that come up.

Depending on which graphics API gets used we also might need to implement our native texture upload system for it.

This would include primarily the graphical client, but once we have the native dependencies compiled, the headless would be implicitly supported as well.

Ideally we'd support both the (now legacy) x64 Macs as well as the new M1 natively. This is dependent on the tooling support though and might take some time.

### Describe alternatives you've considered

Possibly solutions like Proton on OS X if they exist/work well enough.

### Additional context

Splitting this up from the megathread, which became mostly Linux oriented, to separately track and gauge interest in OS X support: https://github.com/Neos-Metaverse/NeosPublic/issues/149

This isn't something that we'd probably prioritize soon and we might wait until we make some changes to the engine /  graphical pipeline first to avoid doing unnecessary work.

Supporting just the headless would be easier as that's mostly the native dependencies, but that's probably way too niche to be worth it, as servers are typically Linux or Windows.

**Comment by peterthethinker** [comment]
Froox.    to be blunt.   Apple is  in the middle of an arch change.   and thusly   there are a TON of powerfull enough  intel units . 

but also the M1 and  rumord M2 are also spicy !  so  are you wanting to support both or ?

**Comment by Frooxius** [comment]
Like I mentioned in the post we'd like to support both x64 and M1 and future ones. However this is more long term, since there's a significant chunk of work and particularly the M1 (and future ones) support requires the tooling we use to support it first.

**Comment by kazu0617** [comment]
@Frooxius Does this Issue include so-called "iOS"? (I looked all over and didn't find an issue that looked like it, and I remembered that .NET itself supports iOS).

**Comment by Frooxius** [comment]
No, iOS support would be a separate issue. This is mostly for the desktop/laptop OS X hardware, while iOS covers iPhone/iPad.

**Comment by kazu0617** [comment]
> 
> 
> No, iOS support would be a separate issue. This is mostly for the desktop/laptop OS X hardware, while iOS covers iPhone/iPad.

I see. I thought I'd make an Issue dedicated to that, so I will.
